pkg/cloudinit/controlplane_init.go

```diff
@@ -27,7 +27,8 @@ const (
 {{template "files" .WriteFiles}}
 runcmd:
 {{- template "commands" .PreK3sCommands }}
-  - 'curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=%s sh -s - server && mkdir -p /run/cluster-api && echo success > /run/cluster-api/bootstrap-success.complete'
+  - {{ if .AirGapped }} "INSTALL_K3S_SKIP_DOWNLOAD=true INSTALL_K3S_EXEC='server' sh /opt/install.sh" {{ else }}'curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=%s sh -s - server' {{ end }}
+  - '{{ .SentinelFileCommand }}'
```

Comment:
use && to append to last command, otherwise sentinel cmd will always execute no matter success or fail